release-24.3: sql/distsql: preserve JobID in redacted logs

spilchen opened the pull request:
Backport 1/1 commits from #146362.

/cc @cockroachdb/release

---

Previously, the JobID log tag could be redacted, making it difficult to trace TTL job execution when viewing redacted logs. This change ensures the JobID is logged as a redact.SafeString, allowing it to appear in redacted outputs for easier debugging.

Additionally, the flowCtx.ID is now also saved as a safe string too.

Epic: none
Release note: none

Release justification: improves job observability

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/146467)
<!-- Reviewable:end -->

yuzefovich reviewed (commented):
Looks like this one is missing the update to FlowCtx.ID in `distsql/server.go`?

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/146467#-:-OPqfEhV1nhTvIrHM7xg:bab910f)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @rafiss)



<!-- Sent from Reviewable.io -->

spilchen reviewed (commented):
This was intentional. Sorry, I should have included more context. I left it off because there were a couple of changes that it depended on that weren’t backported. On this branch, the code to include the flow ID in the log tag is as follows:

```
flowCtx.AmbientContext.AddLogTag("f", flowCtx.ID.Short())
```
  
Where `Short()` is defined as:
 
```
// Short returns the first eight characters of the output of String().
func (u UUID) Short() string {
	return u.String()[:8]
}
```
    
Where as, in 25.1 and newer, `Short()` is defined as:

```
// Short returns an abbreviated version of the UUID containing the first four
// bytes.
func (u UUID) Short() Short {
	return Short{
		b: [shortSize]byte(u[0:shortSize]),
	}
}
``` 

So, I would need to backport #136636 to get the above `Short()` method and #141390 to unredact the Short type.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/146467#-:-OQ3XzS94kJwr5mNYO2G:b-trl2dk)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @rafiss)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
Would this smaller diff work?
```
--- a/pkg/sql/distsql/server.go
+++ b/pkg/sql/distsql/server.go
@@ -368,7 +368,7 @@ func (ds *ServerImpl) setupFlow(
        }
 
        if !f.IsLocal() {
-               flowCtx.AmbientContext.AddLogTag("f", flowCtx.ID.Short())
+               flowCtx.AmbientContext.AddLogTag("f", redact.SafeString(flowCtx.ID.Short()))
                if req.JobTag != "" {
                        flowCtx.AmbientContext.AddLogTag("job", req.JobTag)
                }
```

I'm fine without updating this, so up to you.

spilchen committed:
sql: preserve JobID in redacted logs

Previously, the JobID log tag could be redacted, making it difficult to
trace TTL job execution when viewing redacted logs. This change ensures
the JobID is logged as a redact.SafeString, allowing it to appear in
redacted outputs for easier debugging.

Additionally, the flowCtx.ID is now also saved as a safe string too.

Epic: none
Release note: none

spilchen commented:
That seems simple enough. Let me add that.
